tools/jit/templates/generated_unboxing_wrappers.cpp

```diff
@@ -94,21 +94,24 @@ std::array<bool, N> as_bool_array(const c10::List<bool>& list) {
 }
 
 KernelFunction::InternalBoxedKernelFunction *DUMMY_OPERATION =
-  [](c10::OperatorKernel *, const c10::OperatorHandle &, std::vector<c10::IValue> *) -> void {
+  [](c10::OperatorKernel *, const c10::OperatorHandle &, c10::DispatchKeySet, std::vector<c10::IValue> *) -> void {
     TORCH_CHECK(false, "Operator has been stripped in the custom build.")
   };
 
 class Registerer final {
 public:
-  Registerer&& op(const std::string& schemaStr, KernelFunction::InternalBoxedKernelFunction* boxed_kernel_wrapper) && {
+  Registerer&& op(const std::string& schemaStr,
+                  KernelFunction::InternalBoxedKernelFunction* boxed_kernel_wrapper_old_convention_,
+                  KernelFunction::InternalBoxedKernelFunction* boxed_kernel_wrapper_new_convention_) && {
     static auto& dispatcher = c10::Dispatcher::singleton();
     auto schema = parseSchema(schemaStr);
     schema.setAliasAnalysis(AliasAnalysisKind::FROM_SCHEMA);
     c10::OperatorName name = schema.operator_name();
     RegistrationHandleRAII registration = dispatcher.registerName(name);
     auto op = dispatcher.findOp(name).value();
     registrationHandles_.push_back(std::move(registration));
-    dispatcher.setManuallyBoxedKernelFor_(op, boxed_kernel_wrapper);
+    dispatcher.setManuallyBoxedKernelFor_(op, boxed_kernel_wrapper_old_convention_, false);
+    dispatcher.setManuallyBoxedKernelFor_(op, boxed_kernel_wrapper_new_convention_, true);
```

Comment:
This looks pretty weird.

Reply:
I could probably make it a little cleaner- from the comments at the top of this file its looks like it will go away completely once every op is c10-full, so I didn't think it would last long.

In fact, looking at the `torch/csrc/jit/generated/generated_unboxing_wrapper_everything` now on a new version of master, I don't see any uses of this codegen. I'll double check with Sebastian and see if this file can be taken out entirely- if not, happy to make my change more understandably.

The main issue is that the boxed kernel that this wrapper generates is used by every `KernelFunction` inside of the op. However, not all `KernelFunction`'s in an operator have the same boxed signature anymore (specifically, tracing and autograd get a new boxed signature with an extra 'DispatchKeySet' argument). So now this codegen needs to be aware of the fact that each operator potentially has two boxed function signatures.